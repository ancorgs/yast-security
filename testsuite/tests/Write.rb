# encoding: utf-8

# YaST2: Modules testsuite
#
# Description:
#   Testsuite for the security module
#
# Authors:
#   Michal Svec <msvec@suse.cz>
#
# $Id$
#
# testedfiles: Security.ycp PamSettings.ycp Pam.ycp
module Yast
  class WriteClient < Client
    def main
      Yast.include self, "testsuite.rb"

      Yast.import "Security"

      @m = {
        "CONSOLE_SHUTDOWN"                          => "halt",
        "DISPLAYMANAGER_REMOTE_ACCESS"              => "r4",
        "ENCRYPTION"                                => "md5",
        "kernel.sysrq"                              => "1",
        "FAIL_DELAY"                                => "l2",
        "GID_MAX"                                   => "l3",
        "GID_MIN"                                   => "l4",
        "DISPLAYMANAGER_SHUTDOWN"                   => "r3",
        "PASS_MAX_DAYS"                             => "l7",
        "PASS_MIN_DAYS"                             => "l9",
        "PASS_MIN_LEN"                              => "l10",
        "PASS_WARN_AGE"                             => "l11",
        "PERMISSION_SECURITY"                       => "r5",
        "ROOT_LOGIN_REMOTE"                         => "r6",
        "RUN_UPDATEDB_AS"                           => "r7",
        "UID_MAX"                                   => "l12",
        "UID_MIN"                                   => "l13",
        "SYS_UID_MAX"                               => "l14",
        "SYS_UID_MIN"                               => "l15",
        "SYS_GID_MAX"                               => "l16",
        "SYS_GID_MIN"                               => "l17",
        "USERADD_CMD"                               => "l18",
        "USERDEL_PRECMD"                            => "l19",
        "USERDEL_POSTCMD"                           => "l20",
        "DISABLE_RESTART_ON_UPDATE"                 => "r13",
        "DISABLE_STOP_ON_REMOVAL"                   => "r14",
        "DISPLAYMANAGER_ROOT_LOGIN_REMOTE"          => "r16",
        "DISPLAYMANAGER_XSERVER_TCP_PORT_6000_OPEN" => "r17",
        "net.ipv4.tcp_syncookies"                   => "9",
        "net.ipv4.ip_forward"                       => "10",
        "net.ipv6.conf.all.forwarding"              => "11",
        "SYSTOHC"                                   => "r12",
        "SYSLOG_ON_NO_ERROR"                        => "r15",
        "SMTPD_LISTEN_REMOTE"                       => "r18",
        "DHCPD_RUN_CHROOTED"                        => "r19",
        "DHCPD_RUN_AS"                              => "r20",
        "HIBERNATE_SYSTEM"                          => "r21"
      }

      @E = { "target" => { "bash_output" => {} } }
      @R = {
        "sysconfig" => { "displaymanager" => { "DISPLAYMANAGER" => "" } },
        "target"    => {
          # FileUtils::Exists returns true:
          "stat" => { 1 => 2 }
        }
      }

      Security.Settings = deep_copy(@m)
      Security.modified = true
      TEST(lambda { Security.Write }, [@R, {}, @E], nil)

      nil
    end
  end
end

Yast::WriteClient.new.main
