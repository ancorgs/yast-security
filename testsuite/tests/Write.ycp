/*
 * YaST2: Modules testsuite
 *
 * Description:
 *   Testsuite for the security module
 *
 * Authors:
 *   Michal Svec <msvec@suse.cz>
 *
 * $Id$
 */

{

include "testsuite.ycp";
include "security/io.ycp";

map m = $[
	"CONSOLE_SHUTDOWN"	:	"reboot",
	"CWD_IN_ROOT_PATH"	:	"r2",
	"KDM_SHUTDOWN"		:	"r3",
	"DISPLAYMANAGER_REMOTE_ACCESS" :"r4",
	"PERMISSION_SECURITY"	:	"r5",
	"ROOT_LOGIN_REMOTE"	:	"r6",
	"RUN_UPDATEDB_AS"	:	"r7",
	"FAILLOG_ENAB"		:	"l1",
	"FAIL_DELAY"		:	"l2",
	"GID_MAX"		:	"l3",
	"GID_MIN"		:	"l4",
	"LASTLOG_ENAB"		:	"l5",
	"MD5_SUPPORT"		:	"yes",
	"OBSCURE_CHECKS_ENAB"	:	"l7",
	"PASS_MAX_DAYS"		:	"l8",
	"PASS_MAX_LEN"		:	"l9",
	"PASS_MIN_DAYS"		:	"l10",
	"PASS_MIN_LEN"		:	"l11",
	"PASS_WARN_AGE"		:	"l12",
	"UID_MAX"		:	"l13",
	"UID_MIN"		:	"l14"
];

TEST(``(SecurityWrite(m)),[],nil);

}
