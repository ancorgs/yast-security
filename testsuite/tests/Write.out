Read	.etc.login_defs.FAILLOG_ENAB nil
Write	.etc.login_defs.FAILLOG_ENAB "l1" true
Read	.etc.login_defs.FAIL_DELAY nil
Write	.etc.login_defs.FAIL_DELAY "l2" true
Read	.etc.login_defs.GID_MAX nil
Write	.etc.login_defs.GID_MAX "l3" true
Read	.etc.login_defs.GID_MIN nil
Write	.etc.login_defs.GID_MIN "l4" true
Read	.etc.login_defs.LASTLOG_ENAB nil
Write	.etc.login_defs.LASTLOG_ENAB "l5" true
Read	.etc.login_defs.PASS_MAX_DAYS nil
Write	.etc.login_defs.PASS_MAX_DAYS "l7" true
Read	.etc.login_defs.PASS_MIN_DAYS nil
Write	.etc.login_defs.PASS_MIN_DAYS "l9" true
Read	.etc.login_defs.PASS_WARN_AGE nil
Write	.etc.login_defs.PASS_WARN_AGE "l11" true
Read	.etc.login_defs.UID_MAX nil
Write	.etc.login_defs.UID_MAX "l12" true
Read	.etc.login_defs.UID_MIN nil
Write	.etc.login_defs.UID_MIN "l13" true
Read	.etc.login_defs.SYSTEM_UID_MAX nil
Write	.etc.login_defs.SYSTEM_UID_MAX "l14" true
Read	.etc.login_defs.SYSTEM_UID_MIN nil
Write	.etc.login_defs.SYSTEM_UID_MIN "l15" true
Read	.etc.login_defs.SYSTEM_GID_MAX nil
Write	.etc.login_defs.SYSTEM_GID_MAX "l16" true
Read	.etc.login_defs.SYSTEM_GID_MIN nil
Write	.etc.login_defs.SYSTEM_GID_MIN "l17" true
Read	.etc.login_defs.USERADD_CMD nil
Write	.etc.login_defs.USERADD_CMD "l18" true
Read	.etc.login_defs.USERDEL_PRECMD nil
Write	.etc.login_defs.USERDEL_PRECMD "l19" true
Read	.etc.login_defs.USERDEL_POSTCMD nil
Write	.etc.login_defs.USERDEL_POSTCMD "l20" true
Read	.sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS nil
Write	.sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS "r4" true
Read	.sysconfig.displaymanager.KDM_SHUTDOWN nil
Write	.sysconfig.displaymanager.KDM_SHUTDOWN "r3" true
Read	.sysconfig.locate.RUN_UPDATEDB_AS nil
Write	.sysconfig.locate.RUN_UPDATEDB_AS "r7" true
Read	.sysconfig.security.PERMISSION_SECURITY nil
Write	.sysconfig.security.PERMISSION_SECURITY "r5 local" true
Read	.sysconfig.suseconfig.CWD_IN_ROOT_PATH nil
Write	.sysconfig.suseconfig.CWD_IN_ROOT_PATH "r2" true
Read	.sysconfig.suseconfig.CWD_IN_USER_PATH nil
Write	.sysconfig.suseconfig.CWD_IN_USER_PATH "r2s" true
Read	.sysconfig.sysctl.ENABLE_SYSRQ nil
Write	.sysconfig.sysctl.ENABLE_SYSRQ "yes" true
Write	.etc.login_defs nil true
Write	.sysconfig.displaymanager nil true
Write	.sysconfig.locate nil true
Write	.sysconfig.security nil true
Write	.sysconfig.suseconfig nil true
Write	.sysconfig.sysctl nil true
Write	.etc.inittab.ca ":ctrlaltdel:/sbin/shutdown -r -t 4 now" true
Write	.etc.inittab nil true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_unix2"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_unix2"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Write	.pam."login"."password"."pam_unix" "-md5" true
Write	.pam."login"."password"."pam_pwcheck" "-md5" true
Write	.pam."passwd"."password"."pam_unix" "-md5" true
Write	.pam."passwd"."password"."pam_pwcheck" "-md5" true
Write	.pam."sshd"."password"."pam_unix" "-md5" true
Write	.pam."sshd"."password"."pam_pwcheck" "-md5" true
Write	.pam."rlogin"."password"."pam_unix" "-md5" true
Write	.pam."rlogin"."password"."pam_pwcheck" "-md5" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_unix2"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_unix2"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Write	.pam."login"."password"."pam_unix" "-blowfish" true
Write	.pam."login"."password"."pam_pwcheck" "-blowfish" true
Write	.pam."passwd"."password"."pam_unix" "-blowfish" true
Write	.pam."passwd"."password"."pam_pwcheck" "-blowfish" true
Write	.pam."sshd"."password"."pam_unix" "-blowfish" true
Write	.pam."sshd"."password"."pam_pwcheck" "-blowfish" true
Write	.pam."rlogin"."password"."pam_unix" "-blowfish" true
Write	.pam."rlogin"."password"."pam_pwcheck" "-blowfish" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "no_obscure_checks" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "maxlen=l8" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "minlen=l10" true
Write	.etc.security nil true
Execute	.target.bash "echo 1 > /proc/sys/kernel/sysrq" 0
Execute	.target.bash "/sbin/SuSEconfig --module xdm" 0
Execute	.target.bash "/sbin/SuSEconfig --module kde3" 0
Execute	.target.bash "/sbin/SuSEconfig --module kdm3" 0
Execute	.target.bash "/sbin/SuSEconfig --module permissions" 0
Execute	.target.bash "/sbin/SuSEconfig --module profiles" 0
Return	true
